### PR TITLE
Feat/review api

### PR DESCRIPTION
**Pull request by lluft**

### CODE REVIEW

File: lib/castle-rb/api.rb [hunk 1]
```diff
@@ -26,11 +26,12 @@ def initialize(cookie_id, ip, headers)
...
 
-    def request(endpoint, args)
-      req = Net::HTTP::Post.new(
+    def request(endpoint, args, method: 'Post')
+      http_method = method.to_s.capitalize
+      req = Net::HTTP.const_get(http_method).new(
         "#{Castle.config.api_endpoint.path}/#{endpoint}", @headers)
       req.basic_auth("", Castle.config.api_secret)
-      req.body = args.to_json
+      req.body = args.to_json unless http_method == 'GET'
```
Comment (wallin) [inline lib/castle-rb/api.rb]: This won't match. Either compare with 'Get' or do `http_method.upcase`

File: lib/castle-rb/api.rb [hunk 2]
```diff
@@ -26,11 +26,12 @@ def initialize(cookie_id, ip, headers)
       @headers.delete_if { |k, v| v.nil? }
     end
 
-    def request(endpoint, args)
-      req = Net::HTTP::Post.new(
+    def request(endpoint, args, method: 'Post')
```
Comment (wallin) [inline lib/castle-rb/api.rb]: Style thing: I think we should use symbols to indicate which method, ie. `def request(endpoint, args, method: :post)`, and then do String conversion when doing `const_get`. Looks pretties when using the method

File: lib/castle-rb/api.rb (hunk 1 above)
Comment (lluft) [inline lib/castle-rb/api.rb]: confused `upper_case` with `capitalize`

File: lib/castle-rb/api.rb (hunk 2 above)
Comment (lluft) [inline lib/castle-rb/api.rb]: fixed

**Comment by lluft** [comment]
@wallin incorporated requested changes

**Pull request merged**